readme.md

```diff
@@ -45,10 +45,33 @@ Type: `object`
 
 ##### deep
 
-Type: `boolean`<br>
+Type: `boolean | Function`<br>
 Default: `false`
 
-Recurse nested objects and objects in arrays.
+Recursing behaviour for nested objects and objects in arrays
+
+Deep Value can be boolean:
+
+- If `true`, it recurses all nested objects and objects in arrays.
+
+Or a `DeepKeyFilter` function which has the following properties:
+
+- It has signature `deepKeyFilter(sourceKey: string)`.
+- It must return a `boolean`. Returning `true` states the value related to the currently iterated `sourceKey`
+    shall be recursed if it is a nested object, false otherwise.
+
```

Comment:
I still think the above text could be written more succinctly.

Reply:
I oriented myself when describing the function by the currently existing descriptions of the mapper function.

```
#### mapper

Type: `Function`

Mapping function.

- It has signature `mapper(sourceKey, sourceValue, source)`.
- It must return a two item array: `[targetKey, targetValue]`.
```

But I'll see how it can be reworked ...

Reply:
Ok, I changed it to TS notation. Please see my comment above: https://github.com/sindresorhus/map-obj/pull/24#discussion_r521279876, that it might confuse if there are different signature notation styles.